pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -138,6 +140,17 @@ public static ColumnMetadata fromPropertiesConfiguration(String column, Properti
             .getKeyFor(column, V1Constants.MetadataKeys.Column.TOTAL_NUMBER_OF_ENTRIES));
     builder.setTotalNumberOfEntries(totalNumberOfEntries);
 
+    char paddingCharacter = V1Constants.Str.DEFAULT_STRING_PAD_CHAR;
```

Comment:
Ah, no. You should initialize that to null. It is being fixed in the lines below anyway